runtime/src/extensions.rs

```diff
@@ -439,4 +444,64 @@ mod tests {
 			assert!(result.did_revert());
 		});
 	}
+
+	#[test]
+	#[ignore]
+	fn reading_last_relay_chain_block_number_works() {
+		new_test_ext().execute_with(|| {
+			let _ = env_logger::try_init();
+
+			let (wasm_binary, _) = load_wasm_module::<Runtime>("../contracts/pop-api-examples/read-runtime-state/target/ink/pop_api_extension_demo.wasm").unwrap();
+
+			let init_value = 100;
+
+			let contract = Contracts::bare_instantiate(
+				ALICE,
+				init_value,
+				GAS_LIMIT,
+				None,
+				Code::Upload(wasm_binary),
+				function_selector("new"),
+				vec![],
+				DEBUG_OUTPUT,
+				pallet_contracts::CollectEvents::Skip,
+			)
+				.result
+				.unwrap();
+
+			assert!(
+				!contract.result.did_revert(),
+				"deploying contract reverted {:?}",
+				contract
+			);
+
+			let addr = contract.account_id;
+
+			let function = function_selector("read_relay_block_number");
+			let params = [function].concat();
+
+			let result = Contracts::bare_call(
+				ALICE,
+				addr.clone(),
+				0,
+				Weight::from_parts(100_000_000_000, 3 * 1024 * 1024),
+				None,
+				params,
+				DEBUG_OUTPUT,
+				pallet_contracts::CollectEvents::UnsafeCollect,
+				pallet_contracts::Determinism::Relaxed,
+			);
+
+			if DEBUG_OUTPUT == pallet_contracts::DebugInfo::UnsafeDebug {
```

Comment:
This if statement feels a little unnecessary but maybe I'm not understanding it correctly

Reply:
Agreed. Was something that was copied from existing tests above. Idea seems to be that changing global constant value has no affect on test output.

All these tests need clean up, especially as they rely on the contracts being built locally in the relevant paths and are therefore set as ignored to not break CI checks.

Don't think it warrants further time right now as there are more pressing things to get done.